server/authentication/passportConfig.js

```diff
@@ -0,0 +1,223 @@
+
+// Create new Github strategy.
+passport.use(
+  new GithubStrategy(
+    {
+      clientID: process.env.GITHUB_CLIENT_ID,
+      clientSecret: process.env.GITHUB_CLIENT_SECRET,
+      callbackURL: "/auth/github/callback",
+    },
+    function (accessToken, refreshToken, profile, cb) {
+      const { id, provider, displayName } = profile;
+      const { email, avatar_url } = profile._json;
+      const firstName = "";
+      const lastName = "";
+
+      const user = {
+        displayName: profile.displayName,
+        photos: [{ value: avatar_url }],
+      };
+
+      poolQuery(
+        id,
+        provider,
+        displayName,
+        lastName,
+        firstName,
+        email,
+        avatar_url,
+        user,
+        cb
+      );
+    }
+  )
+);
+
+//     ---------- FACEBOOK AUTHENTICATION ----------     //
+
+// Creates new Facebook strategy
+passport.use(
+  new FacebookStrategy(
+    {
+      clientID: process.env.FACEBOOK_APP_ID,
+      clientSecret: process.env.FACEBOOK_APP_SECRET,
+      callbackURL: "/auth/facebook/callback",
+      profileFields: ["id", "displayName", "email", "name"],
+    },
+    function (accessToken, refreshToken, profile, cb) {
+      const { id, provider, displayName } = profile;
+      const { email, last_name, first_name } = profile._json;
+
+      const fetchUserPicture = fetch(
+        `https://graph.facebook.com/${profile.id}/?fields=picture&type=large&access_token=${accessToken}`
+      )
+        .then((response) => response.json())
+        .then((result) => {
+          const user = {
+            displayName: profile.displayName,
+            photos: [{ value: result.picture.data.url }],
+          };
+
+          poolQuery(
+            id,
+            provider,
+            displayName,
+            last_name,
+            first_name,
+            email,
+            result.picture.data.url,
+            user,
+            cb
+          );
+        })
+        .catch((e) => console.log(e));
+    }
+  )
+);
+
+//     ---------- TWITTER AUTHENTICATION ----------     //
+
+passport.use(
+  new TwitterStrategy(
+    {
+      consumerKey: process.env.TWITTER_CLIENT_ID,
+      consumerSecret: process.env.TWITTER_CLIENT_SECRET,
+      callbackURL: "/auth/twitter/callback",
+    },
+    function (accessToken, refreshToken, profile, cb) {
+      const { id, name, profile_image_url } = profile._json;
+      const { provider } = profile;
+      const email = "";
+      const givenName = "";
+      const familyName = "";
+
+      const user = {
+        displayName: name,
+        photos: [{ value: profile_image_url }],
+      };
+
+      poolQuery(
+        id,
+        provider,
+        name,
+        givenName,
+        familyName,
+        email,
+        profile_image_url,
+        user,
+        cb
+      );
+    }
+  )
+);
+
+//     ---------- HELPERS ----------     //
+
+// Insert new user into db.
+const getDBInsertString = () => {
```

Comment:
It's good practice to declare strings like this as string constants. 

commonly these are capitalised and sit at the top of the file. 

`const DB_INSERT_STRING="INSERT INTO ..." `